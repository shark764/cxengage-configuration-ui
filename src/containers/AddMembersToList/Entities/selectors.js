/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

import { getModalTableItems } from '../../../redux/modules/entities/listItemSelectors';
import { getModalTableUserItems } from '../../../redux/modules/entities/users/selectors';

export const selectModalTableItems = (state, entityName) => {
  switch (entityName) {
    case 'users': {
      return getModalTableUserItems(state, entityName);
    }
    default: {
      return getModalTableItems(state, entityName);
    }
  }
};
