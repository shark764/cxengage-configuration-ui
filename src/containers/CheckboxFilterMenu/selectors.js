import { createSelector } from 'reselect';

const subState = state => state.get('CheckboxFilterMenu');

const menuItems = (state, props) => {
  if (props.menuType) {
    return state.get('CheckboxFilterMenu').get(props.menuType);
  } else {
    return state.get('CheckboxFilterMenu').get('Columns');
  }
};

export const areAllActive = createSelector(
  [subState, menuItems],
  (subState, menuItems) => {
    let active = true;
    menuItems.forEach(item => (!item.get('active') ? (active = false) : null));
    return active;
  }
);

export const selectColumns = createSelector(subState, subState =>
  subState.get('Columns')
);

export const areAllColNotActive = createSelector(subState, subState => {
  let active = false;
  subState
    .get('Columns')
    .forEach(col => (col.get('active') ? (active = true) : null));
  return active;
});

export const selectTimeFormat = createSelector(subState, subState =>
  subState.get('twelveHourFormat')
);
