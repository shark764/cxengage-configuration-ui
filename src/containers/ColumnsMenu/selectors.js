import { createSelector } from 'reselect';

const selectColumnsMenuMap = state => state.get('ColumnsMenu');

export const selectColumnsMenu = createSelector(
  selectColumnsMenuMap,
  ColumnsMenu => ColumnsMenu
);

export const selectColumnsMenuColumns = createSelector(
  selectColumnsMenuMap,
  subState => {
    return subState.get('columns');
  }
);

export const areAllActive = createSelector(selectColumnsMenuMap, subState => {
  let active = true;
  subState
    .get('columns')
    .forEach(col => (!col.get('active') ? (active = false) : null));
  return active;
});

export const areAllNotActive = createSelector(
  selectColumnsMenuMap,
  subState => {
    let active = false;
    subState
      .get('columns')
      .forEach(col => (col.get('active') ? (active = true) : null));
    return active;
  }
);

export const selectTimeFormat = createSelector(selectColumnsMenuMap, subState =>
  subState.get('twelveHourFormat')
);
