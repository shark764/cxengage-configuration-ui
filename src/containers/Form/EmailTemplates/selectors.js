/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

import { Map } from 'immutable';
import {
  getSelectedEntity
} from '../../../redux/modules/crudEndpoint/selectors';

export const getInitialValues = state => {
  const selectedEntity = getSelectedEntity(state);
  const template = selectedEntity.get('template');
  const inherited = selectedEntity.get('inherited');
  let initialValues;
  if (
    template.get('tenantId') ===
    inherited.get('tenantId')
  ) {
    initialValues = {
      email: 'default',
      shared: false,
      subject: inherited.get('subject'),
      body: inherited.get('body')
    };
  } else {
    initialValues = {
      email: 'custom',
      shared: template.get('shared'),
      subject: template.get('subject'),
      body: template.get('body')
    };
  }
  return new Map(initialValues);
}

export const getTemplates = state =>
 getSelectedEntity(state)
  .get('variables')
  .map(variable => variable.get('name'))
  .toJS()
