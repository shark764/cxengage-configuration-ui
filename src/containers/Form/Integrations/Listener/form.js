/*
 * Copyright Â© 2015-2019 Serenova, LLC. All rights reserved.
 */

/**
 *
 * IntegrationListenerForm
 *
 */

import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { DetailHeader, InputField } from 'cx-ui-components';

export default function IntegrationListenerForm({ isSaving, integrationType }) {
  return (
    <Fragment>
      <DetailHeader text="Properties" />
      {integrationType === 'email' && (
        <Fragment>
          <InputField
            name="properties.email"
            label="Email *"
            componentType="input"
            inputType="text"
            data-automation="integrationsListenersEmail"
            disabled={isSaving}
          />
          <InputField
            name="properties.password"
            label="Password *"
            componentType="input"
            inputType="text"
            data-automation="integrationsListenersPassword"
            disabled={isSaving}
            maskValue
          />
        </Fragment>
      )}
      {integrationType === 'facebook' && (
        <Fragment>
          <InputField
            name="properties.pageToken"
            label="Page Token *"
            componentType="input"
            inputType="text"
            data-automation="integrationsListenersPageToken"
            disabled={isSaving}
          />
          <InputField
            name="properties.pageId"
            label="Page ID *"
            componentType="input"
            inputType="text"
            data-automation="integrationsListenersPageId"
            disabled={isSaving}
          />
        </Fragment>
      )}
      {integrationType === 'salesforce' && (
        <Fragment>
          <InputField
            name="properties.topic"
            label="Topic *"
            componentType="input"
            inputType="text"
            data-automation="integrationsListenersTopic"
            disabled={isSaving}
          />
          <InputField
            name="properties.type"
            label="Type *"
            componentType="input"
            inputType="text"
            data-automation="integrationsListenersType"
            disabled={isSaving}
          />
        </Fragment>
      )}
    </Fragment>
  );
}

IntegrationListenerForm.propTypes = {
  initialValues: PropTypes.object,
  isSaving: PropTypes.bool,
  integrationType: PropTypes.string
};
