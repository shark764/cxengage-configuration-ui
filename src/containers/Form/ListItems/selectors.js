/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import { createSelector } from 'reselect';
import {
  getSelectedEntity,
  getSelectedSubEntity
} from '../../../redux/modules/crudEndpoint';

export const getFieldItems = createSelector(getSelectedEntity, selectedList =>
  selectedList.getIn(['listType', 'fields']).toJS()
);

// Removes the first attribute from the fields; it is not editable
export const getUpdateFieldItems = createSelector(
  getSelectedEntity,
  selectedList =>
    selectedList
      .getIn(['listType', 'fields'])
      .shift()
      .toJS()
);

// Removes the uneditable first attribute from the initial values and the key
export const getInitialValues = createSelector(
  getSelectedSubEntity,
  selectedListItem =>
    selectedListItem
      .delete('key')
      .delete(selectedListItem.keySeq().toArray()[0])
);
