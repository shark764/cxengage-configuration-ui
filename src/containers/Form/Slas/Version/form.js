/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

/**
 *
 * InitialVersionForm
 *
 */

import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { InputField, RadioGroupField } from 'cx-ui-components';

export default function InitialVersionForm(props) {
  return (
    <Fragment>
      <InputField
        name="versionName"
        label="Name *"
        data-automation="slasFormFieldVersionName"
        componentType="input"
        inputType="text"
        disabled={props.isSaving || props.viewOnly}
      />
      <InputField
        name="versionDescription"
        label="Description"
        data-automation="slasFormFieldVersionDescription"
        componentType="textarea"
        inputType="text"
        disabled={props.isSaving || props.viewOnly}
      />
      <InputField
        name="slaThreshold"
        label="SLA Threshold"
        data-automation="slasFormFieldSlaThreshold"
        labelHelpText="(seconds)"
        inputType="text"
        disabled={props.isSaving || props.viewOnly}
        componentType="input"
        dataType="number"
      />
      <RadioGroupField
        name="abandonType"
        label="SLA Abandon Type"
        data-automation="slasFormFieldSlaAbandonType"
        disabled={props.isSaving || props.viewOnly}
        options={[
          {
            label: 'Ignore Abandons',
            value: 'ignore-abandons',
            helpText: '(Filter Out / Exclude Short Abandons)'
          },
          {
            label: 'Count Against SLA',
            value: 'count-against-sla',
            helpText: '(Include short abandons against/in SLA)'
          }
        ]}
        required
      />
      {props.slaAbandonType === 'ignore-abandons' && (
        <InputField
          name="abandonThreshold"
          label="Abandon Threshold"
          data-automation="slasFormFieldSlaAbandonThreshold"
          labelHelpText="(seconds)"
          inputType="text"
          disabled={props.isSaving || props.viewOnly}
          componentType="input"
          dataType="number"
        />
      )}
    </Fragment>
  );
}

InitialVersionForm.propTypes = {
  initialValues: PropTypes.object,
  isSaving: PropTypes.bool,
  slaAbandonType: PropTypes.string,
  viewOnly: PropTypes.bool
};
