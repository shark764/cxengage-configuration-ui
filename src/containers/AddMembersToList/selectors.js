/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

import { availableEntitiesForList } from '../../redux/modules/entities/selectors';
import { entitiesMetaData } from '../../redux/modules/entities/metaData';
import { availableItemsForList, availableEntityMembersForList } from '../../redux/modules/entities/listItemSelectors';
import { selectAvailableEntityMembersForList } from '../../redux/modules/entities/users/selectors';

export const selectSidePanelTableItems = (state, currentEntity) => {
  const { dependentEntity } = entitiesMetaData[currentEntity];

  switch (currentEntity) {
    case 'outboundIdentifierLists': {
      return availableEntitiesForList(state, dependentEntity, currentEntity);
    }
    case 'skills':
    case 'groups': {
      return availableItemsForList(state, dependentEntity, currentEntity);
    }
    case 'roles': {
      return availableEntityMembersForList(state, dependentEntity, currentEntity);
    }
    case 'dataAccessReports': {
      return selectAvailableEntityMembersForList(state, ['ASSIGNED_REPORTS_READ']);
    }
    default:
      break;
  }
};
