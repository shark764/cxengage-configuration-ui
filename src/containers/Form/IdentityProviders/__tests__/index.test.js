/*
 * Copyright Â© 2015-2020 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { createStore } from 'redux';
import { getCurrentForm } from '../../../../redux/modules/form/selectors';
import IdentityProvidersForm, { mapStateToProps } from '../';
import { getSelectedEntityId, isCreating, userHasUpdatePermission } from '../../../../redux/modules/entities/selectors';
import {
  selectFormInitialValues,
  formSubmission,
  createFormName
} from '../../../../redux/modules/form/selectors';
import { shallowWithIntl } from '../../../../utils/testUtils';

jest.mock('../../../../redux/modules/entities/selectors');
jest.mock('../../../../redux/modules/form/selectors');
getCurrentForm.mockImplementation(() => 'gets form from state');
getSelectedEntityId.mockImplementation(() => 'mockId');
isCreating.mockImplementation(() => true);
userHasUpdatePermission.mockImplementation(() => true);
selectFormInitialValues.mockImplementation(() => ({ active: true }));

describe('Identity Providers Renders', () => {
  it('renders', () => {
    const store = createStore(state => state);
    const wrapper = (<IdentityProvidersForm store={store}>Child</IdentityProvidersForm>);
    expect(shallowWithIntl(wrapper)).toMatchSnapshot();
  });
});

describe('Maps state to props only using selectors', () => {
  it('validates object created from mapStateToProps', () => {
    expect(mapStateToProps()).toMatchSnapshot();
  });
});

describe('createFormName', () => {
  it('returns proper values', () => {
    expect(createFormName()).toMatchSnapshot();
  });
});

describe('formSubmission', () => {
  const values = {
    name: "mockName",
    description: "mockDescription",
    identityProviderType: "mockType",
    emailMapping: "mockEmailMapping",
    metadataUrl: "mockMetadataUrl",
    identityProvider: "00000000000000000000"
  };
  const dispatch = action => action;
  const props = { dirty: true };
  it('returns proper values', () => {
    expect(formSubmission(values, dispatch, props)).toMatchSnapshot();
  });
});
