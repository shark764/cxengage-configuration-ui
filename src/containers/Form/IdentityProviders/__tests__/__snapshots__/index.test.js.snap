// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Identity Providers Renders renders 1`] = `
<Connect(Connect(ReduxForm))
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  Child
</Connect(Connect(ReduxForm))>
`;

exports[`Maps state to props only using selectors validates object created from mapStateToProps 1`] = `
Object {
  "handleSubmit": [MockFunction],
  "identityProviderTypes": Array [
    Object {
      "label": "URL",
      "value": "url",
    },
    Object {
      "label": "XML File",
      "value": "xml",
    },
    Object {
      "label": "XML Direct Input",
      "value": "xmlDirectInput",
    },
    Object {
      "label": "Shared Access Code",
      "value": "sharedIdentityProviderLinkId",
    },
  ],
  "idpType": undefined,
  "initialValues": Immutable.Map {
    "active": false,
    "emailMapping": "Email",
  },
  "isSaving": true,
  "key": "mockId",
  "userHasUpdatePermission": true,
}
`;

exports[`createFormName returns proper values 1`] = `undefined`;

exports[`formSubmission returns proper values 1`] = `undefined`;
